### ISSUE TITLE
Steam app-type autosyncing erases work done: Store manually fixed app-type values separately

### ISSUE THREAD

**Issue by username0123456789** [bartervg/barter.vg#123]
### Description

Sometime last year (2019) a lot of (likely all) app-type values that had been manually fixed have reset back to the incorrect type or "unknown" etc. It seems all these currently visible apptypes originate from http://store.steampowered.com/api/appdetails?

That source is problematic. For example regional restrictions apply so it won't retrieve info on apps can't be accessed in certain regions, or gives redirected data just like the store page scraping. For example all "Application" apps report as "Game". Most Config and Tool apps can't be retrieved this way, or they often report as "Game", but most of those seem now be "Unknown". Some legit games now have incorrect apptype "Advertising" etc.

Examples of incorrect apptype from store api that were correct before: https://barter.vg/i/3817, https://barter.vg/i/194

### Idea how to do better

Editing the values coming from autosync should not be be the values that are editable, so it should be disabled.

**Create a new _apptype_override_ database entry in which the editors can set the apptype manually.**
**- If apptype_override value exists, then it overrides the value from synced value**
**- If apptype_override value is blank, then the automaticly retrieved apptype is used and assumed to be correct**

If this was done, the incorrect apptypes can then be manually fixed again without having the frustration when syncing erases all the work done later.

**Comment by bartervg** [comment]
Yes, the types were synced with http://store.steampowered.com/api/appdetails and after the `Music` type was introduced, that sync was extended to recheck types even already set. 

In addition to the API inaccuracies, there was an implementation bug that I made that didn't check the appID of the API results. It many cases, it would provide the results for a different API, such providing the info for the related Series for all of the related Episodes apps. 

Yet, the conflict with manual edits and automatic syncing remains.

**Comment by bartervg** [comment]
I added ![image](https://user-images.githubusercontent.com/28576132/80419252-c9371380-88a6-11ea-8399-ba9ee3d58906.png) to prevent manual or automatic updates of the type. The update function will not change the database until unlocked by modifying the UPDATE query to `WHERE id = ? AND locked = 0`. 

Unfortunately, much of the sync code has its own queries that will bypass this, but I edited the ones I found and after a few days, it should be clear if this is happening or not.

**Comment by bartervg** [comment]
Lilly made the following changes that usually would be overwritten, 
https://barter.vg/i/31061/ changed from Unkown to Advertising, locked
https://barter.vg/i/6069/ changed from Advertising to Game, store uses the wrong app id, locked
https://barter.vg/i/51460/ changed from Advertising to Application

**Comment by bartervg** [comment]
The types didn't change so it seems to work. However, it would be useful to know who locked it and when.

**Comment by username0123456789** [comment]
Great job!

I did some testing with some obscure items myself, and noticed if left unlocked they eventually reverted back the wrong type.

If locked, the set value does seem to stay locked now, so far I haven't noticed anything changing.

Fine by me if the changes would be logged on the site, however in any case, I now began keeping a better log of changes I do to the i-pages: 
Comments are of course welcome. There are cases where could probably spend more time pondering. I will start storing timestamps too.
